config/shuttle.go

```diff
@@ -116,32 +117,39 @@ func NewShuttle(appVersion string) *Shuttle {
 			},
 
 			NoLimiter: true,
-			Limits: Limits{
-				SystemLimit: SystemLimit{
-					MinMemory:      1 << 30,
-					MaxMemory:      10 << 30,
-					MemoryFraction: .2,
-
+			Limits: rcmgr.ScalingLimitConfig{
+				SystemBaseLimit: rcmgr.BaseLimit{
+					Memory:          10 << 30,
 					StreamsInbound:  64 << 10,
 					StreamsOutbound: 128 << 10,
 					Streams:         256 << 10,
-
-					ConnsInbound:  256,
-					ConnsOutbound: 256,
-					Conns:         1024,
-
-					FD: 8192,
+					ConnsInbound:    256,
+					ConnsOutbound:   256,
+					Conns:           1024,
+					FD:              8192,
 				},
-				TransientLimit: TransientLimit{
+				TransientBaseLimit: rcmgr.BaseLimit{
+					Memory:          4096,
 					StreamsInbound:  2 << 10,
 					StreamsOutbound: 4 << 10,
 					Streams:         4 << 10,
-
-					ConnsInbound:  256,
-					ConnsOutbound: 256,
-					Conns:         512,
-
-					FD: 1024,
+					ConnsInbound:    256,
+					ConnsOutbound:   256,
+					Conns:           512,
+					FD:              1024,
+				},
+				// can be removed after https://github.com/libp2p/go-libp2p/pull/1878 is released
```

Comment:
Add a `// TODO:`here so we can get back to it once the libp2p PR is merged/released.